spec/lib/manageiq/ssh/util_spec.rb

```diff
@@ -55,6 +55,33 @@ def lastlog
     end
   end
 
+  context "#shell_with_su" do
+    before do
+      @remote_user = 'some_remote_user'
+      @remote_pass = 'some_remote_password'
+      @sudo_user   = 'some_sudo_user'
+      @sudo_pass   = 'some_sudo_password'
+    end
+
+    it "requires remote user and password, as well as sudo user and password" do
+      expect { described_class.shell_with_su(host) }.to raise_error(ArgumentError)
+      expect { described_class.shell_with_su(host, @remote_user, @remote_password) }.to raise_error(ArgumentError)
+    end
+
+    it "requires a block and yields arguments" do
+      expect { |b| described_class.shell_with_su(host, @remote_user, @remote_pass, @sudo_user, @sudo_pass, &b) }.to yield_with_args
```

Comment:
Is this test necessary since the next test also tests that the method yields?

Reply:
Removed.